tensorflow/compiler/xla/literal.cc

```diff
@@ -1725,6 +1726,23 @@ StatusOr<Literal> LiteralBase::BitcastConvert(const Shape& dest_shape) const {
   Literal out(dest_shape);
   std::memcpy(out.root_piece_.buffer(), root_piece().buffer(),
               root_piece().size_bytes());
+
+  // Perform the reshape on little endian encoding even on big endian machines.
+  if (!tensorflow::port::kLittleEndian) {
+    // Swap byte ordering as per the input data type.
+    size_t input_elem_size = ShapeUtil::ByteSizeOfPrimitiveType(
+                              shape().element_type());
+    tensorflow::ByteSwapArray(
```

Comment:
You have to use the result here. Something like 

`TF_RETURN_IF_ERROR(tensorflow::ByteSwapArray(...))`

Reply:
OK, I've updated the code. Please take a look. Thanks!